learning_resources/constants.py

```diff
@@ -234,13 +236,23 @@ class LearningResourceRelationTypes(TextChoices):
 }
 
 
-class LevelType(ExtendedEnum):
-    """Enum for LearningResourceRun levels"""
+class LevelType(NamedEnum):
+    """
+    Enum for LearningResourceRun levels. Each value is a namedtuple with slug
+    and label.
+    """
+
+    _field_names_ = ("slug", "label")
+
+    @staticmethod
+    def pair(label: str) -> tuple[str, str]:
+        slug = re.sub(r"\s+", "_", label.lower()).replace("-", "")
+        return (slug, label)
 
-    undergraduate = "Undergraduate"
-    graduate = "Graduate"
-    high_school = "High School"
-    noncredit = "Non Credit"
-    advanced = "Advanced"
-    intermediate = "Intermediate"
-    introductory = "Introductory"
+    undergraduate = pair("Undergraduate")
+    graduate = pair("Graduate")
+    high_school = pair("High School")
+    noncredit = pair("Non-Credit")
+    advanced = pair("Advanced")
+    intermediate = pair("Intermediate")
+    introductory = pair("Introductory")
```

Comment:
It was suggested in #431 that we construct the filterable API keys programmatically from the label. We don't expose the keys on the API results, but the programmatic construction is predictable.